sechub-wrapper-prepare/src/main/java/com/mercedesbenz/sechub/wrapper/prepare/moduls/PrepareWrapperGIT.java

```diff
@@ -0,0 +1,200 @@
+        } catch (Exception e) {
+            LOG.error("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+            throw new IOException("Error while cleaning git directory: " + pdsPrepareUploadFolderDirectory, e);
+        }
+        waitForProcessToFinish();
+    }
+
+    String getRepositoryURL(String location) {
+        LOG.debug("Prepare location string for private repository");
+        String preFix = "https://" + "$" + PDS_PREPARE_CREDENTIAL_USERNAME + ":" + "$" + PDS_PREPARE_CREDENTIAL_PASSWORD + "@";
+
+        String repositoryURL;
+        if (location.contains("https://")) {
+            repositoryURL = location.replace("https://", preFix);
+        } else if (location.contains("git@")) {
+            location = location.replace(":", "/");
+            repositoryURL = location.replace("git@", preFix);
+        } else if (location.contains("http://")) {
+            repositoryURL = location.replace("http://", preFix);
+        } else if (location.contains("git://")) {
+            repositoryURL = location.replace("git://", preFix);
+        } else {
+            repositoryURL = preFix + location;
+        }
+        return repositoryURL;
+    }
+
+    private void waitForProcessToFinish() {
+
+        /* watching */
+        String watcherThreadName = "PDSPrepareWrapper-git-watcher";
+
+        LOG.debug("Start watcher thread: {}", watcherThreadName);
+
+        StreamDataRefreshRequestWatcherRunnable streamDatawatcherRunnable = new StreamDataRefreshRequestWatcherRunnable();
+        Thread streamDataWatcherThread = new Thread(streamDatawatcherRunnable);
+        streamDataWatcherThread.setName(watcherThreadName);
+        streamDataWatcherThread.start();
+
+        /* waiting for process */
+        LOG.debug("Wait for GIT wrapper to finish process.");
+
+        boolean exitDoneInTime = false;
+        try {
+            exitDoneInTime = process.waitFor(minutesToWaitForResult, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("GIT wrapper could not finish process.", e);
+        }
+
+        if (!exitDoneInTime) {
+            LOG.error("GIT wrapper could not finish process. Waited {} minutes.", minutesToWaitForResult);
+            throw new RuntimeException("GIT wrapper could not finish process. Waited " + minutesToWaitForResult + " minutes.");
+        }
+        streamDatawatcherRunnable.stop();
+        checkProcessExitValue(process.exitValue());
+    }
+
+    private void checkProcessExitValue(int exitValue) {
+        String result = "";
+        if (exitValue == 0) {
+            result = "GIT wrapper finished process successfully.";
+        } else {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getProcess().getErrorStream()))) {
+                StringBuilder stringBuilder = new StringBuilder();
+                String line = null;
+                while (true) {
```

Comment:
I normally try to avoid things like while(true).

A possible solution here could be:

```suggestion
                while ((line = reader.readLine())==null) {
```
or maybe shorter

```suggestion
                while ((line = reader.readLine())) {
```